src/@next/AuthenticationInput/AuthenticationInput.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {
+  ClipboardEvent,
+  RefObject,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { StyledInputsContainer } from './AuthenticationInputStyle';
+import { AuthenticationInputSingle } from './AuthenticationInputSingle';
+
+export interface AuthenticationInputProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  numberOfInputs: number;
+  disabled?: boolean;
+  error?: boolean;
+  forwardedRef?: RefObject<HTMLDivElement>;
+  onChange?: (value: string | null) => void;
+}
+const AuthenticationInputComponent = ({
+  numberOfInputs = 6,
+  error,
+  onChange,
+  forwardedRef,
+  ...rest
+}: AuthenticationInputProps) => {
+  const [inputValues, setInputValues] = useState(
+    new Array<number | ''>(numberOfInputs).fill('')
+  );
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const inputArray = new Array(numberOfInputs).fill('');
+
+  const onChangeRef = useRef(onChange);
```

Comment:
we need to change the refs function on every render, as just incase the onChange handler changes we need that to reflect in the component as well, else we would be calling the outdate on change here


onChangeRef.current = onChange;